mcumgr-core/src/main/java/io/runtime/mcumgr/managers/meta/StatisticsCollector.kt

```diff
@@ -0,0 +1,147 @@
+package io.runtime.mcumgr.managers.meta
+
+import io.runtime.mcumgr.McuMgrCallback
+import io.runtime.mcumgr.exception.McuMgrErrorException
+import io.runtime.mcumgr.exception.McuMgrException
+import io.runtime.mcumgr.managers.StatsManager
+import io.runtime.mcumgr.response.stat.McuMgrStatListResponse
+import io.runtime.mcumgr.response.stat.McuMgrStatResponse
+import java.util.concurrent.atomic.AtomicBoolean
+
+/**
+ * Result of statistic collections.
+ */
+sealed class StatCollectionResult {
+    data class Success(val statistics: Map<String, Map<String, Long>>): StatCollectionResult()
+    data class Cancelled(val statistics: Map<String, Map<String, Long>>): StatCollectionResult()
+    data class Failure(val t: Throwable): StatCollectionResult()
+}
+
+/**
+ * Callback for statistic collections.
+ */
+typealias StatCollectionCallback = (StatCollectionResult) -> Unit
+
+/**
+ * Non-blocking cancellable interface for cancelling an ongoing task.
+ */
+interface Cancellable {
+    fun cancel()
+}
```

Comment:
If you use `Closeable` then you get a lot of Java/Kotlin tools for free (e.g. `resource`-`try`-`catch` and `use`).

Reply:
I dislike having to deal with the throws in the `Closeable`'s `close()` method signature when there is no need to close.

Also, the nomenclature for `cancel` makes more sense than `close` which conveys a lot more information.

Reply:
Good point. If you could throw then `Closeable` is the wrong approach. 👍